src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets

```diff
@@ -73,6 +73,14 @@ Copyright (c) .NET Foundation. All rights reserved.
     </PrepareResourcesDependsOn>
   </PropertyGroup>
 
+  <!-- List of targets that emit DiagnosticMessages items -->
```

Comment:
I agree that as I was implementing this, it became clear that currently only diagnostics that surface in the dependency node really need this infrastructure. Other diagnostics can just Log.LogError|Warning.

On the other hand, a collection of named Diagnostic items that can be manipulated downstream could be useful for as yet unimagined purposes. I'm also wondering if there is any distinction between the coded diagnostics listed in [NuGet](https://github.com/dotnet/cli/blob/d7b0190bd4ce49121a7cd5bdc0423ab85229a085/src/Microsoft.DotNet.ProjectModel/ErrorCodes.NuGet.cs)/[CLI](https://github.com/dotnet/cli/blob/d7b0190bd4ce49121a7cd5bdc0423ab85229a085/src/Microsoft.DotNet.ProjectModel/ErrorCodes.DotNet.cs), and errors/warnings we already emit in our tasks. The former seems dedicated to helping users diagnose why things may not be working as expected, while the latter are more general failures such as a null reference or some other unexpected value. 

I'm open to implementing it either way. Perhaps a compromise is to have an additional property `EmitsDependencyDiagnostic` that is limited to the subset of targets that emit relevant diagnostics.